### PR TITLE
Feature/navbar

### PR DESCRIPTION
**Pull request by vinayak-trivedi**
Date: 10/01/2024

----

## Issue Ticket Number
#5 

## Description

This PR completes the navbar UI for both mobile and desktop version of crypto site

**Documentation Updated?**

-   [ ] Yes
-   [x] No

_If your feature adds a new API, then documentation related to the feature like API-Contracts, and Data-models  must be updated._

**Breaking Changes**

-   [ ] Yes
-   [x] No

_If your feature introduces breaking changes or if something is missing, please mention the related issue tickets._

**Development Tested?**

-   [x] Yes
-   [ ] No

_Confirm whether the changes have been tested locally during development._

**Tested in Staging?**

-   [ ] Yes
-   [x] No

_Indicate whether the changes have been tested in the staging environment._

**Under Feature Flag**

-   [ ] Yes
-   [x] No

_Specify if the changes are currently under a feature flag._

**Database Changes**

-   [ ] Yes
-   [x] No

_Indicate whether the changes include modifications to the database._

### Screenshots

image
image

## Test Coverage

image
+

### CODE REVIEW

File: src/assets/Github_Logo.svg
Comment (vvaibhavdesai) [inline src/assets/Github_Logo.svg:1]: follow a constant naming convention for all assets

File: src/components/GithubLogin/Presentation.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { githubLoginProps } from './githubLogin.types';
+import styles from './githubLogin.module.css';
+import githubLogo from '../../assets/Github_Logo.svg';
+
+const GithubLoginPresentation: React.FC<githubLoginProps> = ({ authUrl }) => {
```
Comment (vvaibhavdesai) [inline src/components/GithubLogin/Presentation.tsx]: you dont need a return here

File: src/components/GithubLogin/githubLogin.util.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { AUTH_URL } from "../../constants/auth";
+
+export const getAuthUrl = () => {
+  let authUrl = AUTH_URL;
+  if (typeof window !== 'undefined') {
```
Comment (vvaibhavdesai) [inline src/components/GithubLogin/githubLogin.util.ts]: add a null check as well

File: src/components/UserProfile/index.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  return (
+    <div data-testid="user_profile" className={styles.user_profile}>
+      <p data-testid="username" className={styles.username}>
+        Hello, {username}
+      </p>
+      <img
+        data-testid="user_image"
+        src={displayPic}
+        className={styles.user_image}
+        alt=""
```
Comment (vvaibhavdesai) [inline src/components/UserProfile/index.tsx]: ???

**Review by vvaibhavdesai — APPROVED** [review]

**Review by satyam73 — APPROVED** [review]

**Pull request merged**